### PR TITLE
Improve code readability by moving fmt args directly into the string

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
There are some of occurrences where I also transformed `write!(f, "{}", x)` into `f.write_str(x.as_str())`.

r? @notriddle

### CODE REVIEW

File: src/librustdoc/clean/utils.rs [hunk 1]
```diff
@@ -618,20 +618,18 @@ pub(super) fn display_macro_source(
...
+        format!("macro_rules! {name} {{\n{}}}", render_macro_arms(cx.tcx, matchers, ";"))
     } else {
         if matchers.len() <= 1 {
             format!(
-                "{}macro {}{} {{\n    ...\n}}",
+                "{}macro {name}{} {{\n    ...\n}}",
                 visibility_to_src_with_space(Some(vis), cx.tcx, def_id),
-                name,
                 matchers.map(|matcher| render_macro_matcher(cx.tcx, matcher)).collect::<String>(),
             )
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/clean/utils.rs:628]: Sure! Did it in a few places but you're right, should have done it for all.

**Comment by GuillaumeGomez** [comment]
Updated!

File: src/librustdoc/doctest.rs [hunk 2]
```diff
@@ -933,7 +933,7 @@ impl Collector {
         if !item_path.is_empty() {
             item_path.push(' ');
         }
-        format!("{} - {}(line {})", filename.prefer_local(), item_path, line)
```
Comment (klensy) [inline src/librustdoc/doctest.rs:936]: There was some discussion that mixing few types of format parameters is harder to read: it need to track different places over code, instead sticking to one method. Or not?

**Comment by GuillaumeGomez** [comment]
Fixed raw string.

@bors r=notriddle

**Pull request merged**